backend/src/packages/auth/auth.app-plugin.ts

```diff
@@ -0,0 +1,67 @@
+import { type FastifyAuthFunction } from '@fastify/auth';
+import { type FastifyReply, type FastifyRequest } from 'fastify';
+import fp from 'fastify-plugin';
+
+import { HttpMessage } from '~/libs/packages/http/http.js';
+import { type UserEntityObjectWithGroupT } from '~/packages/users/users.js';
+
+import { AuthStrategy } from './auth.js';
+import { assertJwtPayload } from './libs/helpers/assert-jwt-payload.helper.js';
+import { createUnauthorizedError } from './libs/helpers/create-unauthorized-error.helper.js';
+import { type AuthPluginOptions } from './libs/types/auth-plugin-options.type.js';
+
+declare module 'fastify' {
+  interface FastifyInstance {
+    [AuthStrategy.VERIFY_JWT]: FastifyAuthFunction;
+  }
+
+  interface FastifyRequest {
+    user: UserEntityObjectWithGroupT;
+  }
+}
+
+const authPlugin = fp<AuthPluginOptions>((fastify, options, done) => {
+  const { userService, jwtService } = options;
+
```

Comment:
Let's add `fastify.decorateRequest('user', null)`?
https://fastify.dev/docs/v2.15.x/Documentation/Decorators/

Reply:
Ah, yes, it was impossible before because of @fastify/jwt, which have been doing this automatically.